src/ToolUpdate/UpdateExecutor.php

```diff
@@ -154,4 +174,35 @@ private function validateHash(string $pathToPhar, ?ToolHash $hash): void
             throw new RuntimeException('Invalid hash for file: ' . $pathToPhar);
         }
     }
-}
\ No newline at end of file
+
+    private function verifySignature(string $pharPath, ToolInformationInterface $tool): ?string
+    {
+        $signatureUrl = $tool->getSignatureUrl();
+        if (null === $signatureUrl) {
+            // TODO: How should we handle missing signatures:
+            // Show a warning or add a configuration how phpcq should deal with it?
+            return null;
+        }
+
+        $signatureName = sprintf('%1$s~%2$s.asc', $tool->getName(), $tool->getVersion());
+        $signaturePath = $this->phpcqPath . '/' . $signatureName;
+        $this->downloader->downloadFileTo($signatureUrl, $signaturePath);
+        $result = $this->verifier->verify(file_get_contents($pharPath),  file_get_contents($signaturePath));
+
+        if ($result->isUntrustedKey() && ($this->untrustedKeyStrategy)($result->getFingerprint(), $tool)) {
```

Comment:
As mentioned above, `untrustedKeyStrategy` handling should be shifted into `verifier` IMO.

Reply:
Done.